app/scripts/modules/core/application/modal/applicationProviderFields.directive.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const angular = require('angular');
+
+module.exports = angular
+  .module('spinnaker.core.application.modal.applicationProviderFields.directive', [
+    require('../../cloudProvider/cloudProvider.registry'),
+    require('../../utils/lodash.js'),
+  ])
+  .directive('applicationProviderFields', function() {
+    return {
+      restrict: 'E',
+      templateUrl: require('./applicationProviderFields.directive.html'),
+      scope: {},
+      bindToController: {
+        application: '=',
+        cloudProviders: '=',
+      },
+      controller: 'ApplicationProviderFieldsCtrl',
+      controllerAs: 'vm',
+    };
+  })
+  .controller('ApplicationProviderFieldsCtrl', function(cloudProviderRegistry) {
+    const templateUrlPath = 'applicationProviderFields.templateUrl';
+
+    this.getRelevantProviderFieldsTemplates = () => {
+      let candidateProvidersToShow;
+
+      if (this.application.cloudProviders.length === 0) {
+        candidateProvidersToShow = this.cloudProviders;
+      } else {
+        candidateProvidersToShow = this.application.cloudProviders;
+      }
+
+      return _(candidateProvidersToShow)
+        .filter(provider => cloudProviderRegistry.hasValue(provider, templateUrlPath))
+        .map(provider => cloudProviderRegistry.getValue(provider, templateUrlPath))
+        .value();
```

Comment:
You should be able to do this without lodash, assuming `candidateProvidersToShow` is an array:
```javascript
return candidateProvidersToShow
  .filter(provider => cloudProviderRegistry.hasValue(provider, templateUrlPath))
  .map(provider => cloudProviderRegistry.getValue(provider, templateUrlPath));
```